externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/TrinoStatement.scala

```diff
@@ -99,50 +99,45 @@ class TrinoStatement(trinoContext: TrinoContext, kyuubiConf: KyuubiConf, sql: St
     val rowBuffer = new ArrayList[List[Any]](MAX_BUFFERED_ROWS)
     var bufferStart = System.nanoTime()
     val result = ArrayBuffer[List[Any]]()
-    try {
-      breakable {
-        while (!dataProcessing.isCompleted) {
-          val atEnd = drainDetectingEnd(rowQueue, rowBuffer, MAX_BUFFERED_ROWS, END_TOKEN)
-          if (!atEnd) {
-            // Flush if needed
-            if (rowBuffer.size() >= MAX_BUFFERED_ROWS ||
-              Duration.fromNanos(bufferStart).compareTo(MAX_BUFFER_TIME) >= 0) {
-              result ++= rowBuffer.asScala
-              rowBuffer.clear()
-              bufferStart = System.nanoTime()
-            }
-
-            val row = rowQueue.poll(MAX_BUFFER_TIME.toMillis, duration.MILLISECONDS)
-            row match {
-              case END_TOKEN => break
-              case null =>
-              case _ => rowBuffer.add(row)
-            }
+
+    breakable {
```

Comment:
can we not use breakable to end a while loop?

Reply:
Sure, use a boolean flag instead.